<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>681</width>
    <height>483</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Filter</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QToolBox" name="toolBox">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>274</width>
             <height>326</height>
            </rect>
           </property>
           <attribute name="label">
            <string>Tags</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QFormLayout" name="formLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Title</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="titleFilter"/>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Artist</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="artistFilter"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QSpinBox" name="minGap">
                <property name="maximum">
                 <number>1000000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;≤ GAP [ms] ≤&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="maxGap">
                <property name="maximum">
                 <number>1000000</number>
                </property>
                <property name="value">
                 <number>1000000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QSpinBox" name="minBPM">
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;center&gt;&lt;p&gt;≤ BPM &amp;le;&lt;/p&gt;&lt;/center&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="maxBPM">
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>1000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="hasVideo">
              <property name="text">
               <string>Has Video</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="hasBackground">
              <property name="text">
               <string>Has Background</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>274</width>
             <height>326</height>
            </rect>
           </property>
           <attribute name="label">
            <string>Validator</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QCheckBox" name="missingSongtextFile">
              <property name="text">
               <string>Missing Songtext</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="missingMandatoryTags">
              <property name="text">
               <string>Missing Mandatory Tags</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="hasUnknownTags">
              <property name="text">
               <string>Unknown Tags</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="missingMP3File">
              <property name="text">
               <string>Missing MP3</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="missingCoverFile">
              <property name="text">
               <string>Missing Cover</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="missingVideoFile">
              <property name="text">
               <string>Missing Video</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="missingBackgroundFile">
              <property name="text">
               <string>Missing Background</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="startsAt0">
              <property name="text">
               <string>Song does not start at 0</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="wellFormed">
              <property name="text">
               <string>Is Well Formed</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="isValid">
              <property name="text">
               <string>Is Valid</string>
              </property>
              <property name="tristate">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_3">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>274</width>
             <height>326</height>
            </rect>
           </property>
           <attribute name="label">
            <string>Properties</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QSpinBox" name="minLen">
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;≤ (MP3) Length [s] ≤&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="maxLen">
                <property name="maximum">
                 <number>1000</number>
                </property>
                <property name="value">
                 <number>1000</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="songScroller">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="songList">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>342</width>
         <height>424</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>681</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFIle">
    <property name="title">
     <string>FIle</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionSources"/>
    <addaction name="actionChecker_Settings"/>
   </widget>
   <addaction name="menuFIle"/>
   <addaction name="menuSettings"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSources">
   <property name="icon">
    <iconset theme="folder">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Songdirs ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionChecker_Settings">
   <property name="text">
    <string>Validator ...</string>
   </property>
  </action>
  <action name="actionRefresh_List">
   <property name="text">
    <string>Refresh List</string>
   </property>
   <property name="toolTip">
    <string>Refreshes the Songlist</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingBackgroundFile</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingCoverFile</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMP3File</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMandatoryTags</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingSongtextFile</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingVideoFile</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasUnknownTags</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startsAt0</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isValid</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wellFormed</sender>
   <signal>clicked()</signal>
   <receiver>actionChecker_Settings</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingSongtextFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMandatoryTags</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasUnknownTags</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMP3File</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingCoverFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingVideoFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingBackgroundFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startsAt0</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wellFormed</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isValid</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>383</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>titleFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>artistFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minGap</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>202</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxGap</sender>
   <signal>valueChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minBPM</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxBPM</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasVideo</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasBackground</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshList()</slot>
 </slots>
</ui>
